### PR TITLE
NWChem: Run dipole moment calculation

### PR DESCRIPTION
**Pull request by ahurta92**
## Description

This pull request allows qcng to run properties calculations.  It also includes changes to harvester.py that reads the dipole moments from nwchem output files.  The PR also includes a test_dipole() function.
## Changelog description

Allows qcng to run properties calculations and parse dipole moments output.

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: qcengine/programs/nwchem/runner.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def parse_output(
             qcvars["CURRENT HESSIAN"] = nwhess
 
         # Normalize the output as a float or list of floats
-        retres = qcvars[f"CURRENT {input_model.driver.upper()}"]
+        retres = qcvars[f"CURRENT ENERGY" ]
```
Comment (loriab) [inline qcengine/programs/nwchem/runner.py]: as-is this will mess up gradient and hessian jobs. want the original line except for properties driver

Comment (ahurta92) [reply qcengine/programs/nwchem/runner.py]: Okay I pushed the change.  I added an if statement for the properties driver

File: qcengine/programs/tests/test_nwchem.py [hunk 2]
```diff
@@ -97,3 +97,40 @@ def test_gradient(nh2):
...
+    assert res["extras"]["qcvars"]["N ATOMS"] == "3"
+    assert res["extras"]["qcvars"]["N BASIS"] == "13"
+
+    # Make sure the properties parsed correctly
+    assert compare_values(-75.764944, res["properties"]["return_energy"], atol=1e-3)
+    assert res["properties"]["calcinfo_natom"] == 3
+    assert res["properties"]["calcinfo_nalpha"] == 6
+    assert res["properties"]["calcinfo_nbasis"] == 13
+    # Make sure Dipole Moment and center of charge parsed correctly
+    assert compare_values(.272949872, float(res["extras"]["qcvars"]["TOTAL DIPOLE MOMENT"]), atol=1e-5)
```
Comment (dgasmith) [inline qcengine/programs/tests/test_nwchem.py]: Can you add the preceding 0 on these numbers?

**Comment by dgasmith** [comment]
Can you rebase this? Test should pass after that.

**Comment by dgasmith** [comment]
@ahurta92 Small heads up, we will be cutting a release soon and would love to get this in!

**Comment by ahurta92** [comment]
Okay great,  I rebased and fixed a bug.  It looks like everything was
approved

**Comment by dgasmith** [comment]
Thanks for patching everything up! @loriab Can you give a final review?

File: qcengine/programs/nwchem/harvester.py [hunk 3]
```diff
@@ -614,11 +614,103 @@ def harvest_outfile_pass(outtext):
...
+                + r".*"
+                + r"\s+"
+                + r"E="
+                + r"([+-]?\s?\d+[.]\d+)"
+                + r"[D]"
+                + r"([+-])"
+                + r"[0]"
+                + r"(\d+)",
+                outtext,
+                re.MULTILINE,
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: black and yapf should both be reproached for their handling of regex

File: qcengine/programs/nwchem/harvester.py [hunk 4]
```diff
@@ -614,11 +614,103 @@ def harvest_outfile_pass(outtext):
...
+            )
+            if mobj:
+                if mobj.group(2) == "+":
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["HOMO"] = np.array([round(lumo, 10)])
+                else:
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["HOMO"] = np.array([round(lumo, 10)])
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: am I missing how the `else` block differs from the `if` block?

File: qcengine/programs/nwchem/harvester.py [hunk 5]
```diff
@@ -614,11 +614,103 @@ def harvest_outfile_pass(outtext):
...
+                outtext,
+                re.MULTILINE,
+            )
+            if mobj:
+                if mobj.group(2) == "+":
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["LUMO"] = np.array([round(lumo, 10)])
+                else:
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["LUMO"] = np.array([round(lumo, 10)])
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: same if/else query

File: qcengine/programs/nwchem/harvester.py [hunk 6]
```diff
@@ -614,11 +614,103 @@ def harvest_outfile_pass(outtext):
...
+    # Search for Center of charge
+    mobj = re.search(
+        r"Center of charge \(in au\) is the expansion point"
+        + r"\n"
+        + r"\s+"
+        + r"X\s+=\s+([+-]?\d+[.]\d+)"
+        + r"\s+"
+        + r"Y\s+=\s+([+-]?\d+[.]\d+)"
+        + r"\s+"
+        + r"Z\s+=\s+([+-]?\d+[.]\d+)",
```
Comment (loriab) [inline qcengine/programs/nwchem/harvester.py]: this is fine if purposeful, but I wanted to make sure you're aware of `NUMBER`, which can simplify reading regex blocks

File: qcengine/programs/nwchem/harvester.py (hunk 4 above)
Comment (ahurta92) [reply qcengine/programs/nwchem/harvester.py]: Sorry, my student and I used this for testing.  I just removed the if-else blocks.

File: qcengine/programs/nwchem/harvester.py (hunk 6 above)
Comment (ahurta92) [reply qcengine/programs/nwchem/harvester.py]: I wouldn't say it's purposeful, I can try changing it to simplify the code.  I wasn't aware of `NUMBER`

Comment (loriab) [reply qcengine/programs/nwchem/harvester.py]: thanks, it'll help readability. fine as a future pass.

File: qcengine/programs/nwchem/harvester.py (hunk 4 above)
Comment (ahurta92) [reply qcengine/programs/nwchem/harvester.py]: Actually just spoke to my student.  At first, he told me it was just for testing but the reason for the IF else is to detect the sign of the power e.g "2.125011D-01",  I'm changing it again.  Sorry about that

**Pull request merged**